### PR TITLE
Traitor chef can use traitor CQC, no multiple defensive stance actions

### LINKED ISSUE #21995: Chef CQC overrides normal CQC.
**Issue Description**:  Chef CQC overrides normal CQC, so when you get normal CQC as chef tot, it doesn't work.

**What did you expect to happen**: Use CQC elsewhere

**What happened instead**: I grabbed random man, and punched him normally.

**Why is this bad/What are the consequences**: I'd like to be able to CQC nerds outside kitchen like a true chef.

**Steps to reproduce the problem**:
1. Be Chef
2. Buy CQC (I was not in kitchen when I bought CQC and applied it to myself)
3. Walla, you are now wondering why your slaps are weak and not epic.

**When did the problem start happening**: I haven't played chef tot much recently, but I remember this happening before as chef tot a couple months ago. However, I was lucky to be recording my chef tot run so its still an issue to this day it seems.

**Extra information**: I did not go to kitchen in-between buying CQC and the fight shown. I did use CQC earlier in the round, before I had bought the normal tot CQC, though I don't know if that contributed to the issue or not.

https://github.com/ParadiseSS13/Paradise/assets/92667559/6c59c87f-30af-48c4-9162-c3c069573749

https://github.com/ParadiseSS13/Paradise/assets/92667559/c5cf26ce-9aeb-4969-b6b7-b70ad7c3f627

### PR DESCRIPTION
**Pull request by tristan1333**
## What Does This PR Do
If a chef learns CQC from the traitor item, he can use it outside of the kitchen. Multiple uses of CQC now doesn't add multiple defence stance actions
Fixes #21995 
## Why It's Good For The Game
Chef should be able to use CQC outside of kitchen when they purchase a CQC manual, multiple defensive stance icons unnecessary 
## Testing
Spawned in as chef, robusted things in the kitchen, made sure couldn't robust outside. Gave chef CQC, could robust where he pleases. Used multiple CQC manuals, didn't get multiple defensive stance icons, CQC still worked, multiple defence stance icons is not good

## Changelog
:cl:
fix: Using CQC manual as chef now allows you to use CQC outside of the kitchen
fix: Multiple uses of CQC manual don't give you multiple defensive stance actions
/:cl:

### CODE REVIEW

File: code/modules/martial_arts/cqc.dm [hunk 1]
```diff
@@ -40,7 +40,16 @@
 
 /datum/martial_art/cqc/teach(mob/living/carbon/human/H, make_temporary)
 	var/datum/action/defensive_stance/defensive = new /datum/action/defensive_stance()
```
Comment (S34NW) [inline code/modules/martial_arts/cqc.dm:42]: Can you move this bit down into the `if(!found)` part?

**Review by S34NW — CHANGES_REQUESTED** [review]

File: code/modules/martial_arts/cqc.dm [hunk 2]
```diff
@@ -39,8 +39,17 @@
 	return ..()
 
 /datum/martial_art/cqc/teach(mob/living/carbon/human/H, make_temporary)
-	var/datum/action/defensive_stance/defensive = new /datum/action/defensive_stance()
-	defensive.Grant(H)
+	var/found = FALSE
+	for(var/datum/martial_art/M in H.mind.known_martial_arts)
+		if(istype(M, /datum/martial_art/cqc/under_siege))
+			M.remove(H)
```
Comment (hal9000PR) [inline code/modules/martial_arts/cqc.dm:45]:
```suggestion
	for(var/datum/martial_art/cqc/under_siege/M in H.mind.known_martial_arts)
		M.remove(H)
```

File: code/modules/martial_arts/cqc.dm [hunk 3]
```diff
@@ -39,8 +39,17 @@
...
+	for(var/datum/action/D in H.actions)
+		if(istype(D, /datum/action/defensive_stance))
+			found = TRUE
+			break
```
Comment (hal9000PR) [inline code/modules/martial_arts/cqc.dm:49]: this part isnt needed as that action will be removed by removing the under_siege martial art

**Review by hal9000PR — DISMISSED** [review]

File: code/modules/martial_arts/cqc.dm (hunk 3 above)
Comment (tristan1333) [reply code/modules/martial_arts/cqc.dm:48]: that's there cos if you use multiple CQC books you get multiple defensive_stance icons

**Review by S34NW — APPROVED** [review]

**Pull request merged**